### ISSUE TITLE
Incompatible with 4.06

### ISSUE THREAD

**Issue by gasche** [andrenth/ocaml-uint#14]
Hi,

The latest version of uint fails on the upcoming OCaml 4.06 release, which enables the separation between (immutable) string and (mutable) byte sequences by default.

```
+ /home/gasche/.opam/4.06.0+local-git-4.06/bin/ocamlfind ocamlc -c -g -annot -bin-annot -I lib -o lib/uint.cmo lib/uint.ml
File "lib/uint.ml", line 109, characters 21-34:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "lib/uint.ml", line 112, characters 35-41:
Error: This expression has type bytes but an expression was expected of type
         string
Command exited with code 2.
```

I will mark past `uint` releases incompatible with the new OCaml release in OPAM.

**Comment by andrenth** [comment]
Hi

I've released 1.2.1 which works on 4.06. Sorry for not merging your pull request, I read the github notification for the issue on my phone and went ahead to fix it without noticing you had opened the pull request too.

**Comment by gasche** [comment]
That's perfectly fine, thanks!
